docs/release-management.md

```diff
@@ -0,0 +1,139 @@
+
+Releasing submariner consists of several manual [steps](https://submariner-io.github.io/contributing/release-process/)
+that need to be carefully executed in order across multiple projects.
+
+This process can be automated as described in the following sections.
+
+## What's a release
+
+   A submariner release is the sum of all components, tagged on a specific set of versions,
+   that work and are tested together.
+   
+   The components include:
+   
+   * subctl
+   * submariner-operator
+   * submariner core (submariner engine, route agent, globalnet ipam)
+   * lighthouse (lighthouse agent, lighthouse coredns)
+   
+  
+## A release repository
+
+   A release repository (submariner-io/releases) can be used to track and automate
+   the whole release process.
+   
+   Yaml files in this repository can be used to control the definition of a release, 
+   
+   ```
+      releases/0.5-<name>/0.5.0-<name>.yaml
+      releases/0.5-<name>/0.5.1-bugfix for....yaml
+      ...
+   ```
+
+## A release yaml
+
+   A release yaml can have the following format:
+   
+   ```yaml
+    ---
+    version: 0.5.0
+    name: the mighty 0.5 release!
+    release-notes: 
+      - bugfixes: >
+        * we fixed this bug
+      - features: >
+        * this feature was added
+        * and this other one
+      - upgrades: >  # optional
+        Upgrade notes important to anybody upgrading from a previous version
+      - deprecations: > # optional
+        Features or settings that have been deprecated and need to be updated
+        by the administrator.
+
+    components:
+      - submariner: <commit-id> 
+      - lighthouse: <commit-id>
+      - submariner-operator: <commit-id>
+      - submariner-charts: <commit-id>
+      - shipyard: <commit-id>
+      - admiral: <commit-id>
+
+   ```
+
+## verification of the release yaml
+
+The verification step would:
+* Verify the format of the yaml file
+
+* Verify that passed CI runs exist for the commit-ids proposed for the components
+
+* Verify that submariner-operator dependencies to projects in go.mod are pointing
+  to the proposed submariner, lighthouse, and shipyard releases, because this affects
+  the behaviour of `subctl verify` to be up-to-date with the e2e tests of those repositories.
+  Otherwise propose a PR for anything missing and stop.
+
+* Verify that projects have Dockerfile.dapper pinned to the specific shipyard release,
+  otherwise send a PR to the projects which didn't pin the image yet and stop.
+
+* Verify that submariner/lighthouse are using the specific admiral commit ID, otherwise send
+  a PR to submariner/lighthouse importing the specific commit ID, and stop.
+
+* Run E2E for the combination of commit-ids with helm & operator (with lighthouse enabled)
+
+## merge of the release yaml
+
+* A script will run in Github actions, executing step by step the tagging and verification
+of the multiple components.
+
+It will retry builds if necessary.
+
+* Shipyard will be tagged for reference
```

Comment:
Don't see a point to do this, since we already expect all repos to be pinned to something which would require the SHA (or tag) to exist before the release.

Also, don't you mean we would tag all the repositories (including Shipyard if we want)?